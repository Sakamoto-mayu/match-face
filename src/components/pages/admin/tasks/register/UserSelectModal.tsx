'use client'
import { ReactNode, useState } from "react"
import { createPortal } from "react-dom"
import WhiteButton from "@/components/ui/button/WhiteButton"
import WhiteCheckButton from "@/components/ui/button/WhiteCheckButton"
import Input from "@/components/ui/Input"
import OrangeButton from "@/components/ui/button/OrangeButton"

const teck = ['Java', 'PHP', 'FR', 'CL', 'ML', 'QA']
const state = ['研修中', '待機中', 'アサイン中']
const groupArr = ['', 'グループ1', 'グループ2', 'グループ3', 'グループ4', 'グループ5', 'グループ6', 'グループ7', 'グループ8']

// ラップしたコンポーネントをModal化するコンポーネントを仮作成
const Modal = ({
  children,
  buttonText,
  canCloseByClickingBackground = true
}: {
  children: ReactNode,
  buttonText: string,
  canCloseByClickingBackground?: boolean,
}) => {
  const [isOpened, setIsOpened] = useState(false)

  const open = () => setIsOpened(true)
  const close = () => setIsOpened(false)

  if (!isOpened) {
    return (
      <WhiteButton label={buttonText} onClick={open} className="text-xs" />
    )
  }

  // レンダリングするDOMをbodyに固定するためPortalを使用
  const elmModal = (
    <div className="fixed top-0 left-0 z-20 flex items-center justify-center w-full h-full">
      <div className="relative z-10 px-10 py-6 w-4/5 max-w-3xl overflow-y-auto bg-white translate-y-3">
        {children}
      </div>
      {canCloseByClickingBackground && <div
        className="absolute top-0 left-0 w-full h-full bg-black opacity-30"
        onClick={close}
      />}
    </div>
  )
  return createPortal(elmModal, document.body)
}

// 実際にレンダリングされるモーダルは以下に記述
const UserSelectModal = () => {
  const [ search, setSearch ] = useState('')
  const [ year, setYear ] = useState('')
  const [ month, setMonth ] = useState('')
  const [ group, setGroup ] = useState('')

  // 2000年からの配列を作成
  const maxYear = (new Date()).getFullYear()
  const yearArr = Array(maxYear-1999)
    .fill(2000)
    .map((num, index) => `${num + index}`)
  yearArr.unshift("")
  
  // 月の配列を作成
  const monthArr = Array(12)
    .fill(1)
    .map((num, index) => `${num + index}`)
  monthArr.unshift("")

  return (
    <Modal buttonText="追加">
      <div>
        <h2>▶️ユーザーを選択する</h2>
      </div>
      <div className="flex flex-col item-center border-2 rounded-md w-11/12 mx-auto mt-2 p-8">
        <div className="flex items-center mb-4 ml-12">
          <Input
            id="search"
            type="text"
            value={search}
            onChange={(e) => setSearch(e.target.value)}
            className="border-light-gray text-xs p-1 w-96"
          />
          <WhiteButton label="検索" className="text-xs ml-4 w-16" />
        </div>
        <div className="flex mb-4 ml-12">
          <select
            name="year"
            id="year"
            onChange={(e) => setYear(e.target.value)}
            value={year}
            className="w-16 border-light-gray"
          >
            {yearArr.map((element, index) => (
              <option key={`year_${index}`} value={element}>{element}</option>
            ))}
          </select>
          <span className="ml-2">年</span>
          <select
            name="month"
            id="month"
            onChange={(e) => setMonth(e.target.value)}
            value={month}
            className="w-20 border-light-gray ml-2"
          >
            {monthArr.map((element, index) => (
              <option key={`month_${index}`} value={element}>{element}</option>
            ))}
          </select>
          <span className="ml-2">月入社</span>
        </div>
        <div className="flex items-center mb-4 ml-10">
          {teck.map((element) => (
            <WhiteCheckButton
              key={element}
              label={element}
              className="text-xs w-16 mx-3"
            />
          ))}
        </div>
        <div className="flex items-center mb-4 ml-10">
          {state.map((element) => (
            <WhiteCheckButton
              key={element}
              label={element}
              className="text-xs w-24 mx-3"
            />
          ))}
        </div>
        <div className="ml-12">
          <h2>▶️グループから検索する</h2>
        </div>
        <div className="ml-12">
          <select
            name="group"
            id="group"
            onChange={(e) => setGroup(e.target.value)}
            value={group}
            className="w-96"
          >
            {groupArr.map((element, index) => (
              <option key={`group_${index}`} value={element}>{element}</option>
            ))}
          </select>
        </div>
        <div className="mx-auto mt-8">
          <OrangeButton label="絞り込み" className="w-28 h-8 text-sm" />
        </div>
      </div>
    </Modal>
  )
}

export default UserSelectModal
