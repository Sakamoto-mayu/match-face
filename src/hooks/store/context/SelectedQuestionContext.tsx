"use client";
import {
  Dispatch,
  ReactNode,
  Reducer,
  createContext,
  useContext,
  useReducer,
} from "react";
import { selectedQuestionReducer } from "../reducer/selectedQuestionReducer";
import { SelectReducerAction } from "@/types/admin/tasks/register/types";

const SelectedQuestionContext = createContext<
  [string[], Dispatch<SelectReducerAction>] | undefined
>(undefined);

export const SelectedQuestionProvider = ({ children }: { children: ReactNode }) => {
  const [state, dispatch] = useReducer<Reducer<string[], SelectReducerAction>>(
    selectedQuestionReducer,
    []
  );
  return (
    <SelectedQuestionContext.Provider value={[state, dispatch]}>
      {children}
    </SelectedQuestionContext.Provider>
  );
};

export const useSelectQuestion = () => {
  const context = useContext(SelectedQuestionContext);
  if (context === undefined)
    throw new Error("Function 'useUserSelect' must be used in Provider.");
  return context;
};
