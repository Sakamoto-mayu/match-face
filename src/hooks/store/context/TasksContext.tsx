"use client";

import React from "react";

// 職種フィルター用context
const JobsFilterContext = React.createContext<
  [string[], React.Dispatch<React.SetStateAction<string[]>>] | undefined
>(undefined);

export const JobsFilterProvider = ({
  children,
}: {
  children: React.ReactNode;
}) => {
  const [filterlist, setFilterList] = React.useState<string[]>([]);
  return (
    <JobsFilterContext.Provider value={[filterlist, setFilterList]}>
      {children}
    </JobsFilterContext.Provider>
  );
};

export const useJobsFilter = () => {
  const context = React.useContext(JobsFilterContext);
  if (context === undefined) {
    throw new Error("useRefine must be used within a Provider");
  }
  return context;
};
// ここまで
