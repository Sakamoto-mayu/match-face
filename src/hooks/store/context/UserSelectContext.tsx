"use client";
import {
  Dispatch,
  ReactNode,
  Reducer,
  createContext,
  useContext,
  useReducer,
} from "react";
import { userSelectReducer } from "../reducer/userSelectReducer";
import { UserSelectReducerAction } from "@/types/admin/tasks/register/types";

const UserSelectContext = createContext<
  [string[], Dispatch<UserSelectReducerAction>] | undefined
>(undefined);

export const CounterProvider = ({ children }: { children: ReactNode }) => {
  const [state, dispatch] = useReducer<Reducer<string[], UserSelectReducerAction>>(
    userSelectReducer,
    []
  );
  return (
    <UserSelectContext.Provider value={[state, dispatch]}>
      {children}
    </UserSelectContext.Provider>
  );
};

export const useUserSelect = () => {
  const context = useContext(UserSelectContext);
  if (context === undefined)
    throw new Error("Function 'useCounter' must be used in Provider.");
  return context;
};
