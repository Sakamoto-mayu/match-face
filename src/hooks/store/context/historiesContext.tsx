'use client'
import { createContext, useContext } from "react";
import { Departments, Skills } from "@/types/admin/histories/admin_histories";
import { skills,departments } from "@/const/admin_histories";

export const SkillsContext = createContext<Skills|undefined>(undefined)
export const DepartmentsContext = createContext<Departments|undefined>(undefined)

// constからのimport(valueの値)をfetchで取得したデータに変える
// const skills = fetch('')
// const departments = fetch('')

export function HistoriesProvider({children}:{children:React.ReactElement}) {
    return (
        <SkillsContext.Provider value = {skills}>
        <DepartmentsContext.Provider value = {departments}>
            {children}
        </DepartmentsContext.Provider>
        </SkillsContext.Provider>
    )
}
export const useSkills = () => {
    const context = useContext(SkillsContext)
    if(context === undefined){
        throw new Error('skillsが見つかりません')
    }
    return context
}
export const useDepartments = () => {
    const context = useContext(DepartmentsContext)
    if(context === undefined){
        throw new Error('departmentsが見つかりません')
    }
    return context
}
