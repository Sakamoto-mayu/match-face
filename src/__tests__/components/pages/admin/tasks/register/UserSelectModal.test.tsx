import UserSelectModal from "@/components/pages/admin/tasks/register/UserSelectModal";
import { UserSelectProvider } from "@/hooks/store/context/UserSelectContext";
import { render, waitFor } from "@testing-library/react";

// コンポーネント内では初期表示用のfetchは無いので、ダミーデータを渡しています

describe("UserSelectModal.tsx", () => {
  it("fetchが呼ばれているかテスト" , async () => {
    global.fetch = jest.fn().mockImplementation((url: string) => {
      if (url.indexOf("departments") !== -1) {
        return {
          ok: true,
          json: async () => ({
            departmentList: [
              { id: 1, name: "部署1" },
              { id: 2, name: "部署2" },
            ],
          }),
        };
      } else if (url.indexOf("statuses") !== -1) {
        return {
          ok: true,
          json: async () => ({
            statusList: [
              { id: 1, name: "ステータス1" },
              { id: 2, name: "ステータス2" },
            ],
          }),
        };
      } else if (url.indexOf("user_groups") !== -1) {
        return {
          ok: true,
          json: async () => ({
            groupList: [
              { id: 1, name: "グループ1" },
              { id: 2, name: "グループ2" },
            ],
          }),
        };
      } else if (url.indexOf("users") !== -1) {
        return {
          ok: true,
          json: async () => ({
            userList: [
              { id: 1, name: "ユーザー1" },
              { id: 2, name: "ユーザー2" },
            ],
          }),
        };
      }
    });

    render(
      <UserSelectProvider>
        {await UserSelectModal()}
      </UserSelectProvider>
    )

    await waitFor(() => {
      expect(fetch).toBeCalledTimes(4)
    })
  })
})
  
