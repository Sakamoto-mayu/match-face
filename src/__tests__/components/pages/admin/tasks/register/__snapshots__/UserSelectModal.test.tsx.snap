// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSelectModal.tsx スナップショットテスト モーダルを開いている状態 1`] = `
<body>
  <div />
  <div
    class="fixed top-0 left-0 z-30 flex items-center justify-center w-full h-full"
  >
    <div
      class="relative z-20 lg:w-3/4 md:w-4/5 w-11/12 p-5 max-w-3xl bg-white"
    >
      <div>
        <h2
          class="text-base ml-3"
        >
          ▶️ユーザーを選択する
        </h2>
      </div>
      <form
        class="flex flex-col justify-center items-center border-2 w-10/12 px-10 py-5 mt-3 mx-auto"
      >
        <div
          class="flex flex-col gap-5 w-fit mx-auto"
        >
          <div
            class="flex items-center"
          >
            <input
              class="border-light-gray border-2 text-xs p-1 w-10/12"
              data-testid="search-box"
              id="search"
              name="search"
              type="text"
              value=""
            />
            <button
              class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm py-2 px-4 rounded-full border shadow-md text-xs ml-4 w-16"
            >
              検索
            </button>
          </div>
          <div
            class="flex items-center gap-1"
          >
            <select
              class="border-2 border-light-gray cursor-pointer text-xs p-1"
              data-testid="select-year"
              id="year"
              name="year"
            >
              <option
                data-testid="year:"
                value=""
              />
              <option
                data-testid="year:2000"
                value="2000"
              >
                2000
              </option>
              <option
                data-testid="year:2001"
                value="2001"
              >
                2001
              </option>
              <option
                data-testid="year:2002"
                value="2002"
              >
                2002
              </option>
              <option
                data-testid="year:2003"
                value="2003"
              >
                2003
              </option>
              <option
                data-testid="year:2004"
                value="2004"
              >
                2004
              </option>
              <option
                data-testid="year:2005"
                value="2005"
              >
                2005
              </option>
              <option
                data-testid="year:2006"
                value="2006"
              >
                2006
              </option>
              <option
                data-testid="year:2007"
                value="2007"
              >
                2007
              </option>
              <option
                data-testid="year:2008"
                value="2008"
              >
                2008
              </option>
              <option
                data-testid="year:2009"
                value="2009"
              >
                2009
              </option>
              <option
                data-testid="year:2010"
                value="2010"
              >
                2010
              </option>
              <option
                data-testid="year:2011"
                value="2011"
              >
                2011
              </option>
              <option
                data-testid="year:2012"
                value="2012"
              >
                2012
              </option>
              <option
                data-testid="year:2013"
                value="2013"
              >
                2013
              </option>
              <option
                data-testid="year:2014"
                value="2014"
              >
                2014
              </option>
              <option
                data-testid="year:2015"
                value="2015"
              >
                2015
              </option>
              <option
                data-testid="year:2016"
                value="2016"
              >
                2016
              </option>
              <option
                data-testid="year:2017"
                value="2017"
              >
                2017
              </option>
              <option
                data-testid="year:2018"
                value="2018"
              >
                2018
              </option>
              <option
                data-testid="year:2019"
                value="2019"
              >
                2019
              </option>
              <option
                data-testid="year:2020"
                value="2020"
              >
                2020
              </option>
              <option
                data-testid="year:2021"
                value="2021"
              >
                2021
              </option>
              <option
                data-testid="year:2022"
                value="2022"
              >
                2022
              </option>
              <option
                data-testid="year:2023"
                value="2023"
              >
                2023
              </option>
            </select>
            <span
              class="text-xs"
            >
              年
            </span>
            <select
              class="border-2 border-light-gray cursor-pointer text-xs p-1"
              data-testid="select-month"
              id="month"
              name="month"
            >
              <option
                data-testid="month:"
                value=""
              />
              <option
                data-testid="month:1"
                value="1"
              >
                1
              </option>
              <option
                data-testid="month:2"
                value="2"
              >
                2
              </option>
              <option
                data-testid="month:3"
                value="3"
              >
                3
              </option>
              <option
                data-testid="month:4"
                value="4"
              >
                4
              </option>
              <option
                data-testid="month:5"
                value="5"
              >
                5
              </option>
              <option
                data-testid="month:6"
                value="6"
              >
                6
              </option>
              <option
                data-testid="month:7"
                value="7"
              >
                7
              </option>
              <option
                data-testid="month:8"
                value="8"
              >
                8
              </option>
              <option
                data-testid="month:9"
                value="9"
              >
                9
              </option>
              <option
                data-testid="month:10"
                value="10"
              >
                10
              </option>
              <option
                data-testid="month:11"
                value="11"
              >
                11
              </option>
              <option
                data-testid="month:12"
                value="12"
              >
                12
              </option>
            </select>
            <span
              class="text-xs"
            >
              月入社
            </span>
          </div>
          <div
            class="flex gap-4 flex-wrap"
          >
            <div>
              <input
                class="peer hidden"
                id="dep_1"
                name="department"
                type="checkbox"
                value="Java"
              />
              <label
                class="text-xs inline-flex items-center justify-center py-2 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs px-1 w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                for="dep_1"
              >
                Java
              </label>
            </div>
            <div>
              <input
                class="peer hidden"
                id="dep_2"
                name="department"
                type="checkbox"
                value="PHP"
              />
              <label
                class="text-xs inline-flex items-center justify-center py-2 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs px-1 w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                for="dep_2"
              >
                PHP
              </label>
            </div>
            <div>
              <input
                class="peer hidden"
                id="dep_3"
                name="department"
                type="checkbox"
                value="FR"
              />
              <label
                class="text-xs inline-flex items-center justify-center py-2 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs px-1 w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                for="dep_3"
              >
                FR
              </label>
            </div>
            <div>
              <input
                class="peer hidden"
                id="dep_4"
                name="department"
                type="checkbox"
                value="CL"
              />
              <label
                class="text-xs inline-flex items-center justify-center py-2 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs px-1 w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                for="dep_4"
              >
                CL
              </label>
            </div>
            <div>
              <input
                class="peer hidden"
                id="dep_5"
                name="department"
                type="checkbox"
                value="ML"
              />
              <label
                class="text-xs inline-flex items-center justify-center py-2 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs px-1 w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                for="dep_5"
              >
                ML
              </label>
            </div>
            <div>
              <input
                class="peer hidden"
                id="dep_6"
                name="department"
                type="checkbox"
                value="QA"
              />
              <label
                class="text-xs inline-flex items-center justify-center py-2 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs px-1 w-16 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                for="dep_6"
              >
                QA
              </label>
            </div>
          </div>
          <div
            class="flex gap-4"
          >
            <div>
              <input
                class="peer hidden"
                id="研修中"
                name="state"
                type="checkbox"
                value="研修中"
              />
              <label
                class="text-xs inline-flex items-center justify-center py-2 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs w-16 px-1 whitespace-nowrap cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                for="研修中"
              >
                研修中
              </label>
            </div>
            <div>
              <input
                class="peer hidden"
                id="待機中"
                name="state"
                type="checkbox"
                value="待機中"
              />
              <label
                class="text-xs inline-flex items-center justify-center py-2 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs w-16 px-1 whitespace-nowrap cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                for="待機中"
              >
                待機中
              </label>
            </div>
            <div>
              <input
                class="peer hidden"
                id="アサイン中"
                name="state"
                type="checkbox"
                value="アサイン中"
              />
              <label
                class="text-xs inline-flex items-center justify-center py-2 bg-white hover:bg-gray-100 border rounded-full shadow-mdtext-xs px-1 w-24 cursor-pointer peer-checked:shadow-sm peer-checked:bg-gray-200 peer-checked:translate-y-0.5"
                for="アサイン中"
              >
                アサイン中
              </label>
            </div>
          </div>
          <div
            class="flex flex-col gap-2"
          >
            <div>
              <h2>
                ▶️グループから検索する
              </h2>
            </div>
            <div>
              <select
                class="border-2 border-light-gray cursor-pointer w-10/12 text-xs p-1"
                data-testid="select-group"
                id="group"
                name="group"
              >
                <option
                  data-testid="group:2023年7月入社"
                  value="2023年7月入社"
                >
                  2023年7月入社
                </option>
                <option
                  data-testid="group:2023年7月入社フロント"
                  value="2023年7月入社フロント"
                >
                  2023年7月入社フロント
                </option>
                <option
                  data-testid="group:テストグループその3"
                  value="テストグループその3"
                >
                  テストグループその3
                </option>
                <option
                  data-testid="group:テストグループその4"
                  value="テストグループその4"
                >
                  テストグループその4
                </option>
              </select>
            </div>
          </div>
        </div>
        <div
          class="mx-auto mt-7"
        >
          <div>
            <button
              class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white rounded-xl p-2 w-40 h-auto text-xs"
              type="submit"
            >
              絞り込み
            </button>
          </div>
        </div>
      </form>
      <div
        class="my-10"
      >
        <table
          class="w-10/12 border-collapse border-2 border-deep-gray text-sm mx-auto"
        >
          <thead
            class="bg-light-gray"
          >
            <tr>
              <th
                class="border-2 border-deep-gray px-4 py-2 text-center w-1/12"
              />
              <th
                class="border-2 border-deep-gray px-4 py-2 text-center w-3/12"
              >
                入社日
              </th>
              <th
                class="border-2 border-deep-gray px-4 py-2 text-center w-20"
              >
                職種
              </th>
              <th
                class="border-2 border-deep-gray px-4 py-2 text-center w-20"
              >
                状態
              </th>
              <th
                class="border-2 border-deep-gray px-4 py-2 text-center w-fit"
              >
                氏名
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                <input
                  class="w-4 h-4 accent-orange cursor-pointer"
                  type="checkbox"
                  value="田中太郎"
                />
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                2023-07-01
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                Java
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                待機中
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                田中太郎
              </td>
            </tr>
            <tr>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                <input
                  class="w-4 h-4 accent-orange cursor-pointer"
                  type="checkbox"
                  value="鈴木二郎"
                />
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                2023-07-01
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                Java
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                待機中
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                鈴木二郎
              </td>
            </tr>
            <tr>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                <input
                  class="w-4 h-4 accent-orange cursor-pointer"
                  type="checkbox"
                  value="伊藤三郎"
                />
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                2023-07-01
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                Java
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                待機中
              </td>
              <td
                class="border-2 border-deep-gray px-4 py-2 text-center"
              >
                伊藤三郎
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="mx-auto mt-8 w-fit"
      >
        <div>
          <button
            class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white rounded-xl p-2 w-40 h-auto text-xs"
          >
            選択完了
          </button>
        </div>
      </div>
    </div>
    <div
      class="absolute top-0 left-0 w-full h-full bg-black opacity-30"
    />
  </div>
  <div>
    <button
      class="bg-white hover:bg-gray-100 active:bg-gray-200 active:translate-y-0.5 active:shadow-sm py-2 px-4 rounded-full border shadow-md text-xs"
    >
      追加
    </button>
  </div>
</body>
`;
